### PR TITLE
Internals doc update after lcache

### PR DESCRIPTION
**Pull request by ktorpi**
Sections after Lcache are updated, sections Property Hashmap, Literal Store and Snapshot are added.

### CODE REVIEW

**Comment by zherczeg** [comment]
LGTM

**Comment by LaszloLango** [comment]
Please rebase and squash.

**Comment by ktorpi** [comment]
Rebased and squashed.

File: 04.internals.md [hunk 1]
```diff
@@ -257,13 +265,13 @@ Several references to single allocated number are not supported. Each reference
 
 ### String
 
-Strings in JerryScript are not just character sequences, but can hold numbers and so-called magic ids too. For common character sequences there is a table in the read only memory that contains magic id and character sequence pairs. If a string is already in this table, the magic id of its string is stored, not the character sequence itself. Using numbers speeds up the property access. These techniques save memory.
+Strings in JerryScript are not just character sequences, but can hold numbers and so-called magic ids too. For common chacter sequences there is a table in the read only memory that contains magic id and chacter sequence pairs. If a string is already in this table, the magic id of its string is stored, not the character sequence itself. Using numbers speeds up the property acces. These techniques save memory.
```
Comment (LaszloLango) [inline 04.internals.md]: These changes are totally wrong

Comment (zherczeg) [inline 04.internals.md]: Why?

**Comment by LaszloLango** [comment]
LGTM

**Pull request merged**